lib/Runtime/Base/FunctionBody.h

```diff
@@ -1658,6 +1660,74 @@ namespace Js
             // same as MachDouble, used in the Func.h
             static const uint DIAGLOCALSLOTSIZE = 8;
 
+            enum class CounterFields : uint8
+            {
+                InterpretedCount                        = 0,
+                VarCount                                = 1,
+                ConstantCount                           = 2,
+                LoopInterpreterLimit                    = 3,
+                OutParamMaxDepth                        = 4,
+                ByteCodeCount                           = 5,
+                ByteCodeWithoutLDACount                 = 6,
+                ByteCodeInLoopCount                     = 7,
+                LoopCount                               = 8,
+                InlineCacheCount                        = 9,
+                RootObjectLoadInlineCacheStart          = 10,
+                RootObjectLoadMethodInlineCacheStart    = 11,
+                RootObjectStoreInlineCacheStart         = 12,
+                IsInstInlineCacheCount                  = 13,
+                ReferencedPropertyIdCount               = 14,
+                ObjLiteralCount                         = 15,
+                LiteralRegexCount                       = 16,
+                InnerScopeCount                         = 17,
+                SavedPolymorphicCacheState              = 18,
+                DebuggerScopeIndex                      = 19,
+
+                // Following counters uses ((uint32)-1) as default value
+                LocalClosureRegister                    = 20,
+                LocalFrameDisplayRegister               = 21,
+                EnvRegister                             = 22,
+                ThisRegForEventHandler                  = 23,
+                FirstInnerScopeRegister                 = 24,
+                FuncExprScopeRegister                   = 25,
+
+                // Signed integers need keep the sign when promoting 
+                SignedFieldsStart                       = 26,
+                SerializationIndex                      = 26,
```

Comment:
SignedFieldsStart  is a marker that from that point, it saves signed value. maybe I can remove this, instead, use a byte in the structure to save the position -- there're 3 bytes at the beginning is not used.